src/components/tables/Details/ShardInformation.tsx

```diff
@@ -0,0 +1,139 @@
+import Editor from '@monaco-editor/react';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import {
+    Accordion,
+    AccordionDetails,
+    AccordionSummary,
+    Alert,
+    Box,
+    Grid,
+    Typography,
+    useTheme,
+} from '@mui/material';
+import { EditorStoreState } from 'components/editor/Store';
+import { PublicationSpecQuery } from 'hooks/usePublicationSpecs';
+import { useRouteStore } from 'hooks/useRouteStore';
+import { useZustandStore } from 'hooks/useZustand';
+import * as monacoEditor from 'monaco-editor/esm/vs/editor/editor.api';
+import { useEffect, useRef, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+import { ShardDetails, shardDetailSelectors } from 'stores/ShardDetail';
+import { ENTITY } from 'types';
+
+interface Props {
+    entityType?: ENTITY.CAPTURE | ENTITY.MATERIALIZATION;
+}
+
+const NEW_LINE = '\r\n';
+
+function ShardInformation({ entityType }: Props) {
+    const theme = useTheme();
+
+    const editorRef = useRef<monacoEditor.editor.IStandaloneCodeEditor | null>(
+        null
+    );
+
+    const [shardDetails, setShardDetails] = useState<ShardDetails | null>(null);
+
+    const shardDetailStore = useRouteStore();
+    const getShardDetails = shardDetailStore(
+        shardDetailSelectors.getShardDetails
+    );
+
+    const specs = useZustandStore<
+        EditorStoreState<PublicationSpecQuery>,
+        EditorStoreState<PublicationSpecQuery>['specs']
+    >((state) => state.specs);
+
+    useEffect(() => {
+        if (specs && specs.length > 0) {
+            setShardDetails(
+                getShardDetails(
+                    specs.find(({ spec_type }) => spec_type === entityType)
+                        ?.catalog_name
+                )
+            );
+        }
+    }, [specs, getShardDetails, setShardDetails, entityType]);
+
+    const handlers = {
+        mount: (editor: monacoEditor.editor.IStandaloneCodeEditor) => {
+            editorRef.current = editor;
+        },
+    };
```

Comment:
Weird thought that never popped into my head until now... I'm not sure we need to create this ref since we don't use it here. I think we might be fine not handling mount when the editors are not used for editing.

Reply:
Follow up PR: #187